include/valkey/cluster.h

```diff
@@ -168,21 +168,23 @@ typedef struct {
     void *tls;
     int (*tls_init_fn)(struct valkeyContext *, struct valkeyTLSContext *);
 
-    /* Synchronous API callbacks */
-    void (*connect_callback)(const valkeyContext *c,
-                             int status);
+    /* Common callbacks. */
     void (*event_callback)(const struct valkeyClusterContext *cc, int event,
                            void *privdata);
     void *event_privdata;
 
+    /* Synchronous API callbacks */
+    void (*connect_callback)(const valkeyContext *c,
+                             int status);
+
     /* Async API event engine adapter.  */
     int (*attach_fn)(valkeyAsyncContext *ac, void *attach_data);
     void *attach_data;
 
     /* Async API callbacks. */
-    valkeyConnectCallback *onConnect;
-    valkeyConnectCallbackNC *onConnectNC;
-    valkeyDisconnectCallback *onDisconnect;
+    valkeyConnectCallback *async_connect_cb;
+    valkeyConnectCallbackNC *async_connect_nc_cb; /* non-const callback */
+    valkeyDisconnectCallback *async_disconnect_cb;
```

Comment:
Good idea. Any thoughts or objections @michael-grunder ?

The const was first removed in hiredis:
```
    Drop `const` on redisAsyncContext in redisConnectCallback
    Since the callback is now re-entrant, it can call apis such as redisAsyncDisconnect()
```
but this change was reverted and the NC variant was added to the API instead:
```
    Maintain backward compatibiliy withour onConnect callback.
    Unfortunately, this is a breaking change we can't make until hiredis v2.0.0.
   
    This commit creates a separate callback member and corresponding
    function that allows us to use the new functionality, while maintaining
    our existing API for legacy code.
```